File: lib/models/rule_rank.dart
/// דירוג כלל — Strong דורס Weak, רק Weak → ambiguous (שליחה ל-AI).
enum RuleRank {
  medium,
  strong,
  weak,
}

extension RuleRankExt on RuleRank {
  static RuleRank fromString(String? s) {
    if (s == null) return RuleRank.medium;
    switch (s.toLowerCase()) {
      case 'strong':
        return RuleRank.strong;
      case 'weak':
        return RuleRank.weak;
      default:
        return RuleRank.medium;
    }
  }
}

File: backend/TheHunterApi/Models/RuleRank.cs
namespace TheHunterApi.Models;

/// <summary>דירוג כלל — Strong דורס Weak, רק Weak → ambiguous (שליחה ל-AI).</summary>
public enum RuleRank
{
    Medium = 0,
    Strong = 1,
    Weak = 2,
}
